contrib/golang/common/go/api/BUILD

```diff
@@ -1,18 +1,59 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library")
+load(
+    "@io_bazel_rules_go//go:def.bzl",
+    "go_library",
+)
+load(
+    "//bazel:envoy_build_system.bzl",
+    "envoy_cc_library",
+    "envoy_contrib_package",
+)
 
 licenses(["notice"])  # Apache 2
 
+envoy_contrib_package()
+
 go_library(
     name = "api",
     srcs = [
+        "api.h",
         "capi.go",
         "cgocheck.go",
         "filter.go",
+        "logger.go",
         "type.go",
     ],
+    cgo = True,
+    clinkopts = select({
+        "@io_bazel_rules_go//go/platform:android": [
+            "-Wl,-unresolved-symbols=ignore-all",
+        ],
+        "@io_bazel_rules_go//go/platform:darwin": [
+            "-Wl,-undefined,dynamic_lookup",
+        ],
+        "@io_bazel_rules_go//go/platform:ios": [
+            "-Wl,-undefined,dynamic_lookup",
+        ],
+        "@io_bazel_rules_go//go/platform:linux": [
+            "-Wl,-unresolved-symbols=ignore-all",
+        ],
+        "//conditions:default": [],
+    }),
     importpath = "github.com/envoyproxy/envoy/contrib/golang/common/go/api",
     visibility = ["//visibility:public"],
     deps = [
         "@org_golang_google_protobuf//types/known/anypb",
     ],
 )
+
+envoy_cc_library(
+    name = "cgo",
```

Comment:
We need to move the C code to a separate directory so the CI can pass.

@doujiang24
Do you have any suggestion about the directory name?

Reply:
maybe `contrib/golang/common/log`? similar to `contrib/golang/common/dso`.